timer/timer.go

```diff
@@ -0,0 +1,120 @@
+package timer
+
+import (
+	"errors"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/nivista/steady/.gen/protos/common"
+	"github.com/nivista/steady/internal/.gen/protos/messaging"
+)
+
+type (
+	// Timer represents the configuration for the execution of a recurring task.
+	Timer struct {
```

Comment:
You will want Timer to be a private struct and have an interface. That way manager which will depend on timer can have a mock for timer.

Reply:
You should have a constructor for this.

Reply:
Yes ok.